### ISSUE TITLE
GPORCA： A volatile quals is pushed down to the subquery unexpectedly when subquery has DISTINCT clause 

### ISSUE THREAD

**Issue by pengzhout** [greenplum-db/gpdb#9871]
GPORCA version: 3.97.1

### Step to reproduce the behavior
This is found in ICW test 'subselect', the simplified repro steps are 
```
create table tenk1 (c1 int, ten int);
create temp sequence ts1;
gpadmin=# explain select * from (select distinct ten from tenk1) ss where ten < 10 + nextval('ts1') order by 1;
                                           QUERY PLAN                                           
------------------------------------------------------------------------------------------------
 Sort  (cost=0.00..431.00 rows=1 width=4)
   Sort Key: ten
   ->  GroupAggregate  (cost=0.00..431.00 rows=1 width=4)
         Group Key: ten
         ->  Sort  (cost=0.00..431.00 rows=1 width=4)
               Sort Key: ten
               ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..431.00 rows=1 width=4)
                     ->  Seq Scan on tenk1  (cost=0.00..431.00 rows=1 width=4)
                           Filter: (ten < (10 + nextval('ts1'::regclass)))
 Optimizer: Pivotal Optimizer (GPORCA) version 3.97.1
(10 rows)
```
the nextval is not expected to been pushed down, there is a discription about the test
```
-- Check that volatile quals aren't pushed down past a DISTINCT: nextval()
-- should not be called more than the nominal number of times
```

### expected behavior
```
gpadmin=# set optimizer to off;
SET
gpadmin=# explain select * from (select distinct ten from tenk1) ss where ten < 10 + nextval('ts1') order by 1;
                                                 QUERY PLAN                                                  
-------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)  (cost=1343.58..1366.08 rows=1000 width=4)
   Merge Key: tenk1.ten
   ->  Sort  (cost=1343.58..1346.08 rows=334 width=4)
         Sort Key: tenk1.ten
         ->  Finalize HashAggregate  (cost=1273.75..1283.75 rows=334 width=4)
               Group Key: tenk1.ten
               Filter: (tenk1.ten < (10 + nextval('ts1'::regclass)))
               ->  Redistribute Motion 3:3  (slice2; segments: 3)  (cost=1176.25..1266.25 rows=1000 width=4)
                     Hash Key: tenk1.ten
                     ->  Partial HashAggregate  (cost=1176.25..1206.25 rows=1000 width=4)
                           Group Key: tenk1.ten
                           ->  Seq Scan on tenk1  (cost=0.00..961.00 rows=28700 width=4)
 Optimizer: Postgres query optimizer
```

A comment from @guofengrichard,
```
In PostgreSQL optimizer, if the subquery uses DISTINCT, volatile quals are not allowed to be pushed down. Not sure how ORCA does that, but the ORCA plan is wrong as it may give incorrect results.
```

**Comment by sambitesh** [comment]
Can we have a repro?

**Comment by pengzhout** [comment]
> Can we have a repro?
updated.

**Comment by kainwen** [comment]
Is this issue fixed by the commit https://github.com/greenplum-db/gpdb/commit/885ca8a9d6bce02ccd6043e94c38f5b58a31a55d ?

**Comment by sambitesh** [comment]
@kainwen yes it is fixed. We can close the issue now

**Comment by sambitesh** [comment]
Issue fixed by 885ca8a
